cli/internal/core/engine.go

```diff
@@ -97,27 +97,34 @@ func (e *Engine) Execute(visitor Visitor, opts EngineExecutionOptions) []error {
 	})
 }
 
-func (e *Engine) getTaskDefinition(taskName string, taskID string) (*Task, error) {
-	pipeline, err := e.getPipelineFromWorkspace(util.RootPkgName)
+func (e *Engine) getTaskDefinition(pkg string, taskName string, taskID string) (*Task, error) {
+	pipeline, err := e.getPipelineFromWorkspace(pkg)
+
 	if err != nil {
+		// Fallback to the root workspace to look for a pipeline if one wasn't
+		// found in the target workspace.
+		if pkg != util.RootPkgName {
+			return e.getTaskDefinition(util.RootPkgName, taskName, taskID)
+		}
+
 		return nil, err
 	}
 
 	if task, ok := pipeline[taskID]; ok {
 		return &Task{
 			Name:           taskName,
-			TaskDefinition: task,
+			TaskDefinition: task.TaskDefinition,
 		}, nil
 	}
 
 	if task, ok := pipeline[taskName]; ok {
 		return &Task{
 			Name:           taskName,
-			TaskDefinition: task,
+			TaskDefinition: task.TaskDefinition,
 		}, nil
 	}
 
-	return nil, fmt.Errorf("Missing task definition, configure \"%s\" or \"%s\" in turbo.json", taskName, taskID)
+	return nil, fmt.Errorf("Could not find \"%s\" or \"%s\" in root config", taskName, taskID)
```

Comment:
This function looks like it can search more than the root config, via `getPipelineFromWorkspace`.